### ISSUE TITLE
No longer compiles with tsc

### ISSUE THREAD

**Issue by jcwatson11** [dex4er/js-promise-duplex#61]
When anything that uses promise-duplex is compiled with the TypeScript compiler (tsc), the following error occurs:

```
node_modules/promise-socket/node_modules/promise-duplex/lib/promise-duplex.d.ts:6:11 - error TS2430: Interface 'DuplexStream' incorrectly extends interface 'Duplex'.
  Property 'closed' is optional in type 'DuplexStream' but required in type 'Duplex'.

6 interface DuplexStream extends Duplex {
            ~~~~~~~~~~~~


Found 1 error.
```

This is because the dependency package promise-duplex (published by the same github user) incorrectly extends the Duplex class in the promise-duplex.ts file.

A child class cannot change the required-ness of a property that exists in a parent class to make it optional when it is required in the parent class. TypeScript requires that child classes must implement the entire contract of a parent class. If a field is required in the parent, it must be required in the child.

**Comment by jcwatson11** [comment]
Also posted in promise-socket, which depends on this project: https://github.com/dex4er/js-promise-socket/issues/77

**Comment by dex4er** [comment]
Fixed in v7.0.0
